zio-flow-server/src/main/scala/zio/flow/server/ZFlowEndpoint.scala

```diff
@@ -0,0 +1,57 @@
+package zio.flow.server
+
+import zhttp.http.HttpError._
+import zhttp.http.Method._
+import zhttp.http._
+import zio._
+import zio.flow._
+import zio.flow.server.ZFlowEndpoint.deserializeFlow
+import zio.schema.codec.JsonCodec
+
+final class ZFlowEndpoint(flowService: ZFlowService) {
+  val endpoint: HttpApp[Any, Nothing] = Http
```

Comment:
This is a good start, some thoughts:

- I would use json responses for all responses (so for example for the not-found error and for returning the `FlowId` from the trigger) so it's easier to write clients for this API
- We will need the polling endpoint from Julien's PR to query running flows, not just to know if they are still running but because we have to be able to get the successful result value or the error
- We should have the ability to define a _parameter type_ in `PUT` - it should be defined by a `Schema` (posted as a serialized `SchemaAst`). It should be optional, but if it is set, we must require a parameter value in the `trigger` endpoint. Passing a parameter to a new flow is not supported currently in the executor but it is easy to add, basically it would set the _environment_ of the flow (and within the flow you can get it with `ZFlow.input[R]`).
- Should have an endpoint that lists all templates